base/dcp_client.go

```diff
@@ -362,8 +364,14 @@ func (dc *DCPClient) openStreamRequest(vbID uint16) error {
 	vbMeta := dc.metadata.GetMeta(vbID)
 
 	options := gocbcore.OpenStreamOptions{}
-	if len(dc.collectionIDs) != 0 {
-		options.FilterOptions = &gocbcore.OpenStreamFilterOptions{CollectionIDs: dc.collectionIDs}
+	// Always use a collection-aware feed if supported
+	if dc.supportsCollections {
+		// If no collection IDs specified, filter to the default collection
+		collIds := dc.collectionIDs
+		if len(collIds) == 0 {
```

Comment:
I don't like the assumption that this will always be 0, I think we should move this logic to `Collection.GetCollectionId` and pass `_default._default` or mark as unsupported.